public/app/features/templating/pickers/OptionsPicker/actions.ts

```diff
@@ -0,0 +1,178 @@
+import debounce from 'lodash/debounce';
+import { StoreState, ThunkDispatch, ThunkResult } from 'app/types';
+import {
+  containsSearchFilter,
+  VariableOption,
+  VariableWithMultiSupport,
+  VariableWithOptions,
+  VariableTag,
+  QueryVariableModel,
+  VariableRefresh,
+} from '../../variable';
+import { toVariablePayload, setCurrentVariableValue } from '../../state/actions';
+import { variableAdapters } from '../../adapters';
+import { getVariable } from '../../state/selectors';
+import { NavigationKey } from '../shared/types';
+import {
+  hideOptions,
+  OptionsPickerState,
+  updateOptionsFromSearch,
+  updateSearchQuery,
+  updateOptionsAndFilter,
+  toggleOption,
+  moveOptionsHighlight,
+  toggleTag,
+} from './reducer';
+import { getDataSourceSrv } from '@grafana/runtime';
+import { getTimeSrv } from 'app/features/dashboard/services/TimeSrv';
+
+export const navigateOptions = (key: NavigationKey, clearOthers: boolean): ThunkResult<void> => {
```

Comment:
@mckn any thoughts on this?

Reply:
yea I agree, the highlighted state feels very transient and something that could be argued should be in component state, but there are the cancel & select and close are real "persisted" actions so maybe it's cleaner to have it all like this here.

Reply:
@dprokop I can see your point and (as far as I know) we have two options:

1. Keep the logic in the component (we started there).
1. Move it to the actions/thunks (where I landed).

The reason for why I preferred this way is the following:

1. I wanted to be able to add tests for this flow since it is pretty important for the picker to properly work. Feels easier to test via thunks/actions then in a component.
1. The code in the component get a bit cleaner and I don't need to import so many dispatchable actions into the component.

If it only where the highlight I totally agree with you regarding keeping that in the component but now we also have select and select + cancel which makes this more like a flow and I would prefer to have those in thunks/actions to make thinks a bit more clear and easy to test.

Does this make sense? Otherwise I am happy to change it. Both solutions work :)